common/src/claims.rs

```diff
@@ -182,11 +234,45 @@ pub struct Claim {
     pub scopes: Vec<Scope>,
     /// The original token that was parsed
     pub(crate) token: Option<String>,
+    /// A struct that holds the account limits.
+    pub limits: Limits,
+    /// The account tier of the subject.
+    pub tier: AccountTier,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub struct Limits {
+    /// The amount of projects this user can create.
+    project_limit: u32,
+}
+
+impl Default for Limits {
+    fn default() -> Self {
+        Self {
+            project_limit: MAX_PROJECTS_DEFAULT,
+        }
+    }
+}
+
+impl Limits {
+    pub fn new(project_limit: u32) -> Self {
+        Self { project_limit }
+    }
+}
+
+pub trait ClaimExt {
+    fn project_limit(&self) -> u32;
+}
+
+impl ClaimExt for Claim {
+    fn project_limit(&self) -> u32 {
+        self.limits.project_limit
+    }
 }
 
 impl Claim {
-    /// Create a new claim for a user with the given scopes
-    pub fn new(sub: String, scopes: Vec<Scope>) -> Self {
+    /// Create a new claim for a user with the given scopes and limits.
+    pub fn new(sub: String, scopes: Vec<Scope>, tier: AccountTier) -> Self {
```

Comment:
Maybe we should add a `limits: Into<Limits>` to communicate the intent better?

Reply:
I originally passed in Limits to this constructor as well, but it seemed redundant, since it's always created from the tier. That may change in the future though, and not having it in the signature does kinda hide it. I can add it back. :+1:

Reply:
Hmm, I started implementing this now, but I'm not sure about this. We will essentially be passing `account_tier` to Claim::new three times: 
```
let claim = Claim::new(name.to_string(), account_tier.into(), account_tier, account_tier);
```
Maybe we should just update the comment to make it clear that this will take a subject, a scope and a tier, and it will create a limits struct from the tier? I don't love that either, but maybe it's good enough.

Reply:
Don't overthink it :)

Reply:
Added it :+1: